components/TwitterContainer.jsx

```diff
@@ -1,9 +1,22 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import TwitterResultCard from '@components/TwitterResult';
 import useFetch from '@hooks/use-fetch';
 
 const Selector = ({ searchStr, resourceType, noRes, noResText }) => {
     const [covidConnectResults, loading] = useFetch({ city: searchStr, materialType: resourceType });
+    const [tweets, setTweets] = useState(covidConnectResults);
+    useEffect(() => {
+        const memo = {};
+        const filteredResults = covidConnectResults.filter(result => {
+            if(!(result.full_text in memo)) {
+                memo[result.full_text] = true;
+                return true;
+            }
+            return false;
+        });
+
+        setTweets(filteredResults);
+    }, [covidConnectResults, loading]);
```

Comment:
const [covidConnectResults, loading] = useFetch({ city: searchStr, materialType: resourceType });
    const removeDuplicateTweets = () => {
        const memo = {};
        const filteredResults = covidConnectResults.filter(result => {
            if(!(result.full_text in memo)) {
                memo[result.full_text] = true;
                return true;
            }
            return false;
        });
        return filteredResults;
    }
    const covidTweetResults = removeDuplicateTweets(covidConnectResults);
    
    return <></>
       
and here,
`
covidConnectResults={covidTweetResults}
`

Instead of having a state and updating it, can we have a the simple filter function before it going to component ?
This is because already while typing, for each letter it renders the component. Having another state inside the component double the render. 🤔

Just my opinion 🤔 Can we do the above ?👆